### PR TITLE
add cuda compute 8.7 to Cmakelists.txt to support Nvidia Orin devices

### PR DESCRIPTION
**Pull request by jywu-msft**
add cuda compute 8.7 to Cmakelists.txt to support Nvidia Orin devices
this is to enable JetPack 5.0 which supports Orin based devices.

### CODE REVIEW

**Review by snnn — APPROVED** [review]

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -1801,6 +1801,7 @@ if (onnxruntime_USE_CUDA)
       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_53,code=sm_53") # TX1, Nano
       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_62,code=sm_62") # TX2
       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_72,code=sm_72") # AGX Xavier, NX Xavier
+      set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_87,code=sm_87") # AGX Orin, NX Orin
```
Comment (hariharans29) [inline cmake/CMakeLists.txt:1804]: Just curious - does this mean the binary size of the onnxruntime .dll and .so will now increase as there is now support for one more arch and possibly more PTX code (due to the `gencode` flag) ?

Comment (snnn) [reply cmake/CMakeLists.txt:1804]: No. These are controlled by Azure Devops Yaml files.

Comment (hariharans29) [reply cmake/CMakeLists.txt:1804]: Ok, those are for the release binaries, but for anyone building from source (with no options provided) ?

Comment (jywu-msft) [reply cmake/CMakeLists.txt:1804]: yes, that's right. someone building from source on a jetson device doesn't need to add extra args.

**Comment by jywu-msft** [comment]
@snnn please re-review, I forgot I should add a cuda version check.

**Review by chilo-ms — APPROVED** [review]

**Pull request merged**